File: ListView/app/src/main/java/com/example/listview/MainActivity.java
package com.example.listview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
ListView lstView;
ArrayList<String> arrNames = new ArrayList<>();
int[] arrNo = new int[]{1,2,3,4,5};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        lstView = findViewById(R.id.lstView);

    }
}

File: MyNotesApp/app/src/main/java/com/virasinfotech/mynotesapp/MainActivity.kt
package com.virasinfotech.mynotesapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.virasinfotech.mynotesapp.database.Note
import com.virasinfotech.mynotesapp.databinding.ActivityMainBinding
import com.virasinfotech.mynotesapp.viewmodels.MainActivityViewModel

class MainActivity : AppCompatActivity() {

    private val binding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }
     lateinit var noteList : List<Note>
    lateinit var viewModel: MainActivityViewModel
    lateinit var adapter: RvAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        noteList = ArrayList()

        viewModel = ViewModelProvider(this).get(MainActivityViewModel::class.java)

        var observer = Observer<List<Note>>{
            noteList =it

            adapter = RvAdapter(noteList,this)
            binding.rv.layoutManager= LinearLayoutManager(this)
            binding.rv.adapter =adapter
        }
        viewModel.notesList.observe(this,observer)

        adapter = RvAdapter(noteList,this)
        binding.rv.layoutManager= LinearLayoutManager(this)
        binding.rv.adapter =adapter

        binding.floatingActionButton.setOnClickListener {
            var intent = Intent(this@MainActivity,AddEditActivity::class.java)
            startActivity(intent)
        }
    }
}